x/authz/simulation/genesis.go

```diff
@@ -1,35 +1,59 @@
 package simulation
 
 import (
-	"encoding/json"
-	"fmt"
 	"math/rand"
 
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
 	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
 	"github.com/cosmos/cosmos-sdk/x/authz"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 )
 
-// GenAuthorizationGrant returns an empty slice of authorization grants.
-func GenAuthorizationGrant(_ *rand.Rand, _ []simtypes.Account) []authz.GrantAuthorization {
-	return []authz.GrantAuthorization{}
+// genAuthorizationGrant returns a slice of authorization grants.
+func genAuthorizationGrant(r *rand.Rand, accounts []simtypes.Account) []authz.GrantAuthorization {
+	authorizations := make([]authz.GrantAuthorization, len(accounts)-1)
+	for i := 0; i < len(accounts)-1; i++ {
+		granter := accounts[i]
+		grantee := accounts[i+1]
+		authorizations[i] = authz.GrantAuthorization{
+			Granter:       granter.Address.String(),
+			Grantee:       grantee.Address.String(),
+			Authorization: generateRandomGrant(r),
+		}
+	}
+
+	return authorizations
+}
+
+func generateRandomGrant(r *rand.Rand) *codectypes.Any {
+	authorizations := make([]*codectypes.Any, 2)
+	authorizations[0] = newAnyAuthorization(banktypes.NewSendAuthorization(sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(1000)))))
+	authorizations[1] = newAnyAuthorization(authz.NewGenericAuthorization(sdk.MsgTypeURL(&govtypes.MsgSubmitProposal{})))
+
+	return authorizations[r.Intn(len(authorizations))]
+}
+
+func newAnyAuthorization(grant authz.Authorization) *codectypes.Any {
```

Comment:
```suggestion
func newAnyAuthorization(a authz.Authorization) *codectypes.Any {
```

Reply:
Let's don't mix authorization with grant. Grant is a structure which wraps Authorizations.